strax/processing/peak_splitting.py

```diff
@@ -202,8 +198,6 @@ def _split_hitlets(split_finder, peaks, orig_dt, is_split, min_area,
                 r = new_hits[offset]
                 r['time'] = h['time'] + prev_split_i * h['dt']
                 r['channel'] = h['channel']
-                # Hitlet specific
-                r['record_i'] = h['record_i']
```

Comment:
Daniel, please correct me if I am wrong, but we might as well delete the `_split_hitlets` function, the two are now exactly the same apart from the `'max_gap'` line in peaks, which is not computed anyway. I think we can very simply clear up all this duplication

Reply:
Here is a diff:
![afbeelding](https://user-images.githubusercontent.com/22295914/116529135-e3059f00-a8dc-11eb-8972-355180130f0b.png)

Reply:
Hmm okay, just looked again at https://github.com/AxFoundation/strax/pull/309 and remembered the pain

Reply:
Perhaps, partially to make our lives easy, and on the other hand since I foresee a genuine advantage and no clear loss, why not add `max_goodness_of_split` to hitlets?

Reply:
Yes, but same comment as above. I did not want to add another thing to make things more complicated :D Adding those fields is not any problem since we are copying in the end data from temp_hitlets_with_data_field to hitlets. So every field we add is only temporarily.

Reply:
okay, however you decide. I would at this point not be distracted by making things more simple by replacing a function by an already in place function which is tested.

But I see your point. On the other hand, just go for it 😄